### PR TITLE
Fix the WPT export script

### PR DESCRIPTION
**Pull request by mrobinson**
- Have the WPT exporter script use the WPT_SYNC_TOKEN for checking out
  wpt.
- Make sure the local WPT repository is unshallow when pushing.
- When searching for existing PRs use the main GitHub search API,
  as the pull request search does not seem to properly process
  the "head" parameter.
- When deleting branches in the downstream WPT repository, use
  the full URL to avoid trying to modify the upstream repository.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes OR

### CODE REVIEW

File: .github/workflows/upstream-wpt-changes.yml [hunk 1]
```diff
@@ -25,7 +25,11 @@ jobs:
         with:
           path: wpt
           repository: 'web-platform-tests/wpt'
-          token: ${{ secrets.GITHUB_TOKEN }}
+          # The token here must be the token that we will use to push to the
+          # WPT repository and not the token used for GitHub actions, because
+          # the checkout action sets up an extraheader authorization override
```
Comment (delan) [inline .github/workflows/upstream-wpt-changes.yml:30]: nit: typo in “extraheader”?

File: etc/ci/upstream-wpt-changes/wptupstreamer/step.py [hunk 2]
```diff
@@ -198,7 +204,11 @@ def run(self, run: SyncRun):
         self.name += f":{run.sync.downstream_wpt.get_branch(self.branch_name)}"
         logging.info("  -> Removing branch used for upstream PR")
         if not run.sync.suppress_force_push:
-            run.sync.local_wpt_repo.run("push", "origin", "--delete",
+            user = run.sync.github_username
+            token = run.sync.github_api_token
+            repo = run.sync.downstream_wpt_repo
+            remote_url = f"https://{user}:{token}@github.com/{repo}.git"
+            run.sync.local_wpt_repo.run("push", remote_url, "--delete",
```
Comment (delan) [inline etc/ci/upstream-wpt-changes/wptupstreamer/step.py:211]: will the token in this command be redacted by github actions?

**Review by delan — APPROVED** [review]
Thanks! Looks good, with one question.

File: .github/workflows/upstream-wpt-changes.yml (hunk 1 above)
Comment (mrobinson) [reply .github/workflows/upstream-wpt-changes.yml]: This git setting is actually called `extraheader`. I can add quotes here to make it clearer though and also link to the GitHub actions issue.

**Comment by mrobinson** [comment]
@bors-servo r=delan

@delan Thank you for the review!

File: etc/ci/upstream-wpt-changes/wptupstreamer/step.py (hunk 2 above)
Comment (delan) [reply etc/ci/upstream-wpt-changes/wptupstreamer/step.py:211]: sounds good!

File: .github/workflows/upstream-wpt-changes.yml (hunk 1 above)
Comment (delan) [reply .github/workflows/upstream-wpt-changes.yml]: yes please!

**Pull request merged**